### ISSUE TITLE
Introduce `WidgetReportError` component and use in widgets

### ISSUE THREAD

**Issue by felixarntz** [google/site-kit-wp#2670]
Following up on #2252, #2612, and #2613, we should also pass a scoped, widget-specific version of the `ReportError` component to every widget component as prop. While it will not receive any special handling like e.g. `ReportZero` for now, this brings it inline with the other components used by widgets and allows third-party widgets to use this component for errors in the same way. We could alternatively make `ReportError` publicly available, but that would result in an inconsistent API and prevent us from adding special widget-specific handling to the component in the future.

---------------

_Do not alter or remove anything below. The following sections will be managed by moderators only._

## Acceptance criteria

* There should be a new `WidgetReportError` component in `assets/js/googlesitekit/widgets/components`.
    * It should require `widgetSlug`, `moduleSlug`, and `error` props.
    * It should always just return a `ReportError` element.
    * _Any_ props other than `widgetSlug` should be passed through to the returned `ReportError`.
    * While `widgetSlug` is currently unused, it still should be required for parity with the other components and to allow us to add widget-specific handling in the future.
* The `getWidgetComponentProps` function should be modified to also include a scoped version of `WidgetReportError` (with `widgetSlug` already provided) in the returned object that will be passed to every registered widget component.
* Every widget that currently returns an imported `ReportError` should instead return the `WidgetReportError` from props. The same props as before should be passed (`widgetSlug` isn't necessary since `props.WidgetReportError` is the wrapped version that already has it).

## Implementation Brief

* Create a new `WidgetReportError` component in the `assets/js/googlesitekit/widgets/components` directory:
  * It should require `widgetSlug`, `moduleSlug` and `error` props;
  * It should return a `ReportError` element with all props passed to the `WidgetReportError` component itself except the `widgetSlug`.
* Update the `getWidgetComponentProps` function to include a scoped version of the `WidgetReportError` component into the returning object;
* Update the following widgets to sue the scoped `WidgetReportError` component instead of `ReportError`:
  * AdSense:
    * `DashboardSummaryWidget`
    * `DashboardTopEarningPagesWidget`
  * Analytics:
    * `DashboardAllTrafficWidget`
    * `DashboardBounceRateWidget`
    * `DashboardGoalsWidget`
    * `DashboardPopularPagesWidget`
    * `DashboardUniqueVisitorsWidget`
    * In the new All Traffic widget use the incoming `WidgetReportError` component in the parent component itself and don't pass it to the children `UserCountGraph` and `TotalUserCount` components.
      * Once #2429 is merged into develop, update the `LegacyDashboardAllTraffic` component to pass the non-scoped `ReportError` element to the `DashboardAllTrafficWidget` element as a fallback for the scoped version.
  * Search Console:
    * `DashboardClicksWidget`
    * `DashboardImpressionsWidget`
    * `DashboardPopularKeywordsWidget`

### Test Coverage

* Update existing tests if any is failing due to the aforementioned changes.

### Visual Regression Changes

* No visual changes are expected as a result of this change.

## QA Brief

* Make sure that all widgets that currently use the `ReportError` component should be updated to use a `WidgetReportError` element provided via props.

## Changelog entry

* Introduce `WidgetReportError` component, which should be used by widgets to display an overall error state.

**Comment by felixarntz** [comment]
@eugene-manuilov IB is almost good to go, except one thing:
> In the new All Traffic widget pass the incoming `WidgetReportError` component to the children `UserCountGraph` and `UserDimensionsPieChart` components to use it there.

That shouldn't be done, `WidgetReportError` should only be used where the entire widget returns an error since only in that case the overall error state should be set for the widget. This relates to what we've decided for #2675: Only for the `ReportError` returned by the entire widget (i.e. in the root component), `props.WidgetReportError` should be used. The child components can keep using their imported `ReportError`s.

**Comment by eugene-manuilov** [comment]
Yep, IB is updated.

**Comment by felixarntz** [comment]
IB ✅

**Comment by tofumatt** [comment]
QA ✅
